### PR TITLE
Deployment actions & scraper migration

### PR DESCRIPTION
**Pull request by justkahdri**
# Pull Request
Add new workflow that triggers a deployment on staging (after launch it should point to `meilisearch.com/main`) and then runs the docs scraper.

### CODE REVIEW

File: docs-scraper.config.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          "hierarchy_radio_lvl5",
+          "hierarchy_radio_lvl0",
+          "hierarchy_lvl1",
+          "hierarchy_lvl2",
+          "hierarchy_lvl3",
+          "hierarchy_lvl4",
+          "hierarchy_lvl5",
+          "hierarchy_lvl0",
+          "objectID",
+          "anchor"
```
Comment (maryamsulemani97) [inline docs-scraper.config.json:43]:
```suggestion
          "*"
```
We want to use the current docs search settings

**Review by dichotommy — CHANGES_REQUESTED** [review]
Please update the `MEILISEARCH_HOST_URL` and `MEILISEARCH_API_KEY` secret variables.

File: docs-scraper.config.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+              "relevant"
+          ]
+      }
+  },
+  "min_indexed_level": 2,
+  "selectors_exclude": [
+      "#page-navigation",
+      "#feedback"
+  ]
+}
```
Comment (guimachiavelli) [inline docs-scraper.config.json:51]: New line character after the last line to make github happy

```suggestion
}

```

**Review by dichotommy — APPROVED** [review]
LGTM!

File: docs-scraper.config.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{
+  "index_uid": "new-docs-structure",
```
Comment (dichotommy) [inline docs-scraper.config.json:2]:
```suggestion
  "index_uid": "new-docs-staging",
```

**Pull request merged**